crates/services/sync/src/import.rs

```diff
@@ -98,6 +99,26 @@ pub struct Import<P, E, C> {
     executor: Arc<E>,
     /// Consensus port.
     consensus: Arc<C>,
+    /// A cache of already validated header or blocks.
+    cache: SharedMutex<HashMap<Range<u32>, CachedData>>,
```

Comment:
suggestion: Perhaps we should consider using a [`DashMap`](https://crates.io/crates/dashmap) here instead like in https://github.com/FuelLabs/fuel-core/pull/2352

Reply:
Alternatively since we're storing ranges, a `BTreeMap` could be useful. I'm a bit hesitant towards using `Range<u32>` as key here. It's not obvious to me that we'll process exactly the same ranges if the stream is restarted. It would be more robust to instead maintain a map of the block height to the cached data at that height.

Or can we be sure that we're
1. Not storing overlapping ranges and
2. will query the same ranges that we have cached?

Reply:
I can use `DashMap` didn't know about it, seems interesting.

About storing range, I answered here: https://github.com/FuelLabs/fuel-core/pull/2361#issuecomment-2417970089 , however I'm trying to implement it now but the problem is that p2p only works with range and so it's a problem if we have a cache in the middle of the range

Reply:
Yeah that's why I think a `BTreeMap` is pretty nice since they allow for querying ranges of keys directly.

Reply:
YYeah but this doesn't solve the problem that we will have to split ranges asked to p2p and multiply the number of requests

Reply:
DashMap isn't an option anymore as we are using BTreeMap